src/main/scala/it/pagopa/interop/agreementprocess/service/CatalogManagementService.scala

```diff
@@ -24,57 +24,51 @@ object CatalogManagementService {
   def validateCreationOnDescriptor(eservice: EService, descriptorId: UUID): Future[Unit]   = {
     val allowedStatus: List[EServiceDescriptorState] =
       List(EServiceDescriptorState.PUBLISHED, EServiceDescriptorState.SUSPENDED)
-    validateEServiceDescriptorStatus(eservice, descriptorId, allowedStatus, true)
+    validateEServiceDescriptorStatus(eservice, descriptorId, allowedStatus)
   }
   def validateSubmitOnDescriptor(eservice: EService, descriptorId: UUID): Future[Unit]     = {
     val allowedStatus: List[EServiceDescriptorState] =
       List(EServiceDescriptorState.PUBLISHED, EServiceDescriptorState.SUSPENDED)
-    validateEServiceDescriptorStatus(eservice, descriptorId, allowedStatus, true)
+    validateEServiceDescriptorStatus(eservice, descriptorId, allowedStatus)
   }
 
   def validateEServiceDescriptorStatus(
```

Comment:
there is no more need of this function, you can just call 
- `validateDescriptor` for `validateActivationOnDescriptor` 
- `validateLatestDescriptor` for `validateCreationOnDescriptor` and `validateSubmitOnDescriptor`

Reply:
I thought about replacing `validateEServiceDescriptorStatus`, that would be my next request.
Next time, if I can, I'll replace it right away